.github/workflows/deploy.yml

```diff
@@ -36,9 +36,12 @@ jobs:
       - name: Deploy
         run: yarn deploy
 
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v1
+        with:
+          path: 'dist'
+
       - name: Deploy to github pages 🚀
         if: github.ref == 'refs/heads/master'
-        uses: JamesIves/github-pages-deploy-action@v4.4.2
-        with:
-          branch: gh-pages # The branch the action should deploy to.
-          folder: build # The folder the action should deploy.
+        id: deployment
+        uses: actions/deploy-pages@v2
```

Comment:
I'm not sure to understand, the difference between the two

"Upload artifact" is uploading the `dist` folder. But then what is doing the `actions/deploy-pages@v2`?

Reply:
Thanks for the explaination